internal/controller/operatorconfigmap_controller.go

```diff
@@ -950,3 +1002,71 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+// TODO: Remove it in 4.20 release
+func (c *OperatorConfigMapReconciler) getCsiCustomTolerations() ([]corev1.Toleration, []corev1.Toleration, error) {
+	rookCephOperatorConfigName, csiPluginTolerationKey, csiProvisionerTolerationKey := "rook-ceph-operator-config", "CSI_PLUGIN_TOLERATIONS", "CSI_PROVISIONER_TOLERATIONS"
+	nodePluginTolerations, controllerPluginTolerations := []corev1.Toleration{}, []corev1.Toleration{}
+
+	// list the rook ceph cm
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	rookCephOperatorConfig.Name = rookCephOperatorConfigName
+	rookCephOperatorConfig.Namespace = c.OperatorNamespace
+
+	if err := c.get(rookCephOperatorConfig); err != nil {
+		if kerrors.IsNotFound(err) {
+			return nodePluginTolerations, controllerPluginTolerations, nil
+		}
+		return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to get Rook Ceph Operator ConfigMap: %v", err)
+	}
+
+	// Get the CSI tolerations from the Rook Ceph Operator ConfigMap
+	csiPluginTolerations, plugin := rookCephOperatorConfig.Data[csiPluginTolerationKey]
+	csiProvisionerToleration, provisioner := rookCephOperatorConfig.Data[csiProvisionerTolerationKey]
+
+	if plugin && csiPluginTolerations != "" {
+		tolerations, err := parseTolerations(csiPluginTolerations)
+		if err != nil {
+			return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to parse csi plugin tolerations: %v", err)
+		}
+		nodePluginTolerations = tolerations
+	}
+
+	if provisioner && csiProvisionerToleration != "" {
+		tolerations, err := parseTolerations(csiProvisionerToleration)
+		if err != nil {
+			return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to parse csi provisioner tolerations: %v", err)
+		}
+		controllerPluginTolerations = tolerations
+	}
+
+	return nodePluginTolerations, controllerPluginTolerations, nil
+}
+
+func parseTolerations(csiPluginTolerations string) ([]corev1.Toleration, error) {
+	if csiPluginTolerations == "" {
+		return []v1.Toleration{}, nil
+	}
+
+	rawJSON, err := yaml.ToJSON([]byte(csiPluginTolerations))
+	if err != nil {
+		return []v1.Toleration{}, err
+	}
+
+	var tolerations []v1.Toleration
+	err = json.Unmarshal(rawJSON, &tolerations)
+	if err != nil {
+		return []v1.Toleration{}, err
+	}
+
+	for i := range tolerations {
+		if tolerations[i].Key == "" {
+			tolerations[i].Operator = corev1.TolerationOpExists
+		}
+
+		if tolerations[i].Operator == corev1.TolerationOpExists {
+			tolerations[i].Value = ""
+		}
+	}
```

Comment:
sry, I'm failing to understand this logic, what is the intent, better add a comment?

Reply:
I am too, I didnt understand this,
I pointed out here https://github.com/red-hat-storage/ocs-client-operator/pull/411#discussion_r2156548584
Confirm with @Madhu-1 we can keep the code as it is

Reply:
This was a upstream code where some details are missing, we are adding it. its pretty old code.